umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/iso19115_2.clj

```diff
@@ -242,185 +242,188 @@
-               [:gmd:extent
-                [:gml:TimeInstant {:gml:id (su/generate-id)}
-                 [:gml:timePosition date]]]]])]]
-         [:gmd:processingLevel
-          [:gmd:MD_Identifier
-           [:gmd:code (char-string (-> c :ProcessingLevel :Id))]
-           [:gmd:codeSpace (char-string processing-level-code-space-string)]
-           [:gmd:description (char-string (-> c :ProcessingLevel :ProcessingLevelDescription))]]]]]
-       (sdru/generate-service-related-url (:RelatedUrls c))
-       (aa/generate-content-info-additional-attributes additional-attributes)
-       [:gmd:contentInfo
-        [:gmd:MD_ImageDescription
-         [:gmd:attributeDescription ""]
-         [:gmd:contentType ""]
-         [:gmd:processingLevelCode
-          [:gmd:MD_Identifier
-           [:gmd:code (char-string (-> c :ProcessingLevel :Id))]
-           [:gmd:codeSpace (char-string processing-level-code-space-string)]
-           [:gmd:description (char-string (-> c :ProcessingLevel :ProcessingLevelDescription))]]]]]
-       (let [related-url-distributions (sdru/generate-distributions c)
-             data-center-distributors (data-contact/generate-distributors (:DataCenters c))]
+             [:gco:CharacterString (extent-description-string c)]]
+            (spatial/generate-zone-identifier c)
+            (spatial/spatial-extent-elements c)
+            (spatial/generate-vertical-domain c)
+            (spatial/generate-orbit-parameters c)
+            (for [temporal (:TemporalExtents c)
+                  rdt (:RangeDateTimes temporal)]
+              [:gmd:temporalElement
+               [:gmd:EX_TemporalExtent
+                [:gmd:extent
+                 [:gml:TimePeriod {:gml:id (su/generate-id)}
+                  [:gml:beginPosition (:BeginningDateTime rdt)]
+                  (if (:EndsAtPresentFlag temporal)
+                    [:gml:endPosition {:indeterminatePosition "now"}]
+                    [:gml:endPosition (su/nil-to-empty-string (:EndingDateTime rdt))])]]]])
+            (for [temporal (:TemporalExtents c)
+                  date (:SingleDateTimes temporal)]
+              [:gmd:temporalElement
+               [:gmd:EX_TemporalExtent
+                [:gmd:extent
+                 [:gml:TimeInstant {:gml:id (su/generate-id)}
+                  [:gml:timePosition date]]]]])]]
+          (when processing-level
+            [:gmd:processingLevel
+             [:gmd:MD_Identifier
+               [:gmd:code (char-string (:Id processing-level))]
+               [:gmd:codeSpace (char-string processing-level-code-space-string)]
+               [:gmd:description (char-string (:ProcessingLevelDescription processing-level))]]])]]
+      (sdru/generate-service-related-url (:RelatedUrls c))
+      (aa/generate-content-info-additional-attributes additional-attributes)
+      [:gmd:contentInfo
+       [:gmd:MD_ImageDescription
+        [:gmd:attributeDescription ""]
+        [:gmd:contentType ""]
+        (when processing-level
```

Comment:
Thank you - I had trouble finding why this passed in my branch, but not in the main build.